### PR TITLE
chore: bump react-native to 0.64

### PR DESCRIPTION
**Pull request by tido64**
## Summary

Bumps react-native to 0.64. This only affects maintainers.

## Test Plan

CI should be green.

### CODE REVIEW

File: android/src/test/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpersTest.kt [hunk 1]
```diff
@@ -1,79 +0,0 @@
-package com.reactnativecommunity.asyncstorage.next
```
Comment (Krizzu) [inline android/src/test/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpersTest.kt:1]: why have you removed those tests?

File: android/src/test/java/com/reactnativecommunity/asyncstorage/next/TestUtils.kt [hunk 2]
```diff
@@ -1,84 +0,0 @@
-package com.reactnativecommunity.asyncstorage.next
```
Comment (Krizzu) [inline android/src/test/java/com/reactnativecommunity/asyncstorage/next/TestUtils.kt:1]: This is used by `ArgumentHelpersTest.kt`, why removing this?

File: android/src/test/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpersTest.kt (hunk 1 above)
Comment (tido64) [reply android/src/test/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpersTest.kt:1]: These are only testing `createNativeCallArguments`, and `createNativeCallArguments` is a test-only utility. They're not testing real code as far as I can see. Did I miss anything?

Comment (Krizzu) [reply android/src/test/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpersTest.kt:1]: Might seem like it, but these test are actually checking [.toEntryList](https://github.com/react-native-async-storage/async-storage/blob/master/android/src/main/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpers.kt#L8) and  [.toKeyList](https://github.com/react-native-async-storage/async-storage/blob/master/android/src/main/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpers.kt#L33) extension functions

Comment (tido64) [reply android/src/test/java/com/reactnativecommunity/asyncstorage/next/ArgumentHelpersTest.kt:1]: Aha! Let me restore them then.

File: android/src/test/java/com/reactnativecommunity/asyncstorage/next/TestUtils.kt [hunk 3]
```diff
@@ -1,84 +1,19 @@
...
+import com.facebook.react.bridge.JavaOnlyArray
 import com.facebook.react.bridge.ReadableArray
-import com.facebook.react.bridge.ReadableMap
-import com.facebook.react.bridge.ReadableType
-import com.facebook.react.bridge.WritableArray
 
-fun <T : Any> createNativeCallArguments(vararg list: T): ReadableArray {
-    val args = ArrayList(list.toList())
-    return TestingArguments(args)
+fun List<Any?>.toReadableArray(): ReadableArray {
```
Comment (Krizzu) [inline android/src/test/java/com/reactnativecommunity/asyncstorage/next/TestUtils.kt:6]: Nice one 👍 
You could actually move this extension to `ArgumentHelpersTest.kt` file, to remove this one

**Review by Krizzu — APPROVED** [review]
This looks awesome, well done 👍

File: android/src/test/java/com/reactnativecommunity/asyncstorage/next/TestUtils.kt (hunk 3 above)
Comment (tido64) [reply android/src/test/java/com/reactnativecommunity/asyncstorage/next/TestUtils.kt]: Yeah, I think that makes sense.

**Pull request merged**